src/renderer/components/cluster-manager/cluster-actions.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import { clusterSettingsURL } from "../+cluster-settings";
+import { landingURL } from "../+landing-page";
+
+import { clusterStore } from "../../../common/cluster-store";
+import { broadcastMessage, requestMain } from "../../../common/ipc";
+import { clusterDisconnectHandler } from "../../../common/cluster-ipc";
+import { ConfirmDialog } from "../confirm-dialog";
+import { Cluster } from "../../../main/cluster";
+
+const navigate = (route: string) =>
+  broadcastMessage("renderer:navigate", route);
+
+/**
+ * Creates handlers for high-level actions
+ * that could be performed on an individual cluster
+ * @param cluster Cluster
+ */
+export const ClusterActions = (cluster: Cluster) => ({
+  showSettings: () => navigate(clusterSettingsURL({
+    params: { clusterId: cluster.id }
+  })),
```

Comment:
This doesn't seem to work correctly. (This also didn't work before). But calling this code from within the iframe the `clusterSettingsURL` doesn't exist so we are redirected to `/` or the landing page.

It would be great if once the user exited out of the settings the view would go back to what they were doing before.

Reply:
This seems to have been fixed in #2330